AndelaOSP/Andela-Socials
# feat(slack):connect with slack

#### What Does This PR Do?
It enables the project to connect to slack and get a user's oauth token
#### Description Of Task To Be Completed
- Add slack authorization endpoint
- Get slack user's authorization code and use it to fetch their oauth token
- Save user token to db 
#### Any Background Context You Want To Provide?
None
#### How can this be manually tested?
- Ensure you have the following .env variables set with valid values:
```
SLACK_BOT_TOKEN
SLACK_CLIENT_ID
SLACK_CLIENT_SECRET
```
- Navigate to `/api/v1/slack/authorize` (in a browser) to launch to the slack authorization process
- Log into django-admin
- Use the code received from slack as input for the `createUserAuth` mutation
#### What are the relevant pivotal tracker stories?
[#165718383](https://www.pivotaltracker.com/story/show/165718383)
